core/deployment/src/main/java/io/quarkus/deployment/execannotations/ContainerInvokedMethodPredicateBuildItem.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.deployment.execannotations;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * Carries a predicate that allows identifying methods that can are invoked by the container
+ * (or, in fact, by any of the frameworks present in the container).
+ * <p>
+ * Used to detect wrong usage of annotations that may change the execution semantics of methods
+ * ({@code @Blocking}, {@code @NonBlocking}, {@code @RunOnVirtualThread}), as these annotations
+ * are implemented directly by the various frameworks and may only be put to "entrypoint" methods.
+ * Placing these annotations on methods that cannot be invoked by the container is always wrong.
+ */
+public final class ContainerInvokedMethodPredicateBuildItem extends MultiBuildItem {
```

Comment:
I think that the name of this build item is too generic and does not indicate what exactly the predicate does. It should probably also contain words like "execution", "model", "annotations" :shrug:.

Reply:
I would personally prefer if this were called `managed` instead of `container`.

Also, in JAX-RS / Jakarta REST, `container` unfortunately means `server`  which only adds to the confusion

Reply:
Renamed to `ExecutionModelAnnotationsAllowedBuildItem`.

Container here was meant very generally (like, Quarkus in its entirety, or more precisely the current composition of frameworks/extensions/containers in any given application), but that's void now I guess.

Reply:
Got rid of all occurences of the word "container".